app/src/main/java/org/xbmc/kore/ui/sections/settings/SettingsFragment.java

```diff
@@ -217,4 +233,76 @@ public boolean onPreferenceClick(Preference preference) {
             }
         });
     }
+
+    private void setupLanguagePreference(final ListPreference languagePref) {
+        Locale[] locales = getLocales(BuildConfig.SUPPORTED_LOCALES);
+
+        Arrays.sort(locales, new Comparator<Locale>() {
+            @Override
+            public int compare(Locale o1, Locale o2) {
+                return o1.getDisplayName().compareToIgnoreCase(o2.getDisplayName());
+            }
+        });
+
+        String[] displayNames = new String[locales.length];
+        String[] entryValues = new String[locales.length];
+        for(int index = 0; index < locales.length; index++) {
+            Locale locale = locales[index];
+            displayNames[index] = locale.getDisplayName(locale);
+            entryValues[index] = getLanguageCountryCode(locale);
+        }
+
+        languagePref.setValue(getLanguageCountryCode(getCurrentLocale()));
+        languagePref.setEntries(displayNames);
+        languagePref.setEntryValues(entryValues);
+        languagePref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object o) {
+                languagePref.setValue(o.toString());
+                updatePreferredLanguage(o.toString());
+                return true;
+            }
+        });
+    }
+
+    private String getLanguageCountryCode(Locale locale) {
+        String result = locale.getLanguage();
+        if (!locale.getCountry().isEmpty()) {
+            result += locale.getCountry();
```

Comment:
Missing the `-`... ex: pt-BR

Now that i look into it, is it a `-` or a `_`? I believe it's a `-`. One more reason to use the default functions

Reply:
Nice catch. Fixed.